tests/unit/httpd_test.cc

```diff
@@ -2000,17 +2000,17 @@ SEASTAR_THREAD_TEST_CASE(test_http_with_broken_wire) {
 }
 
 future<> test_client_close_connection(bool chunked) {
-    return async([] {
+    return async([chunked] {
         loopback_connection_factory lcf(1);
-        auto make_test_request = [&lcf]() {
+        auto make_test_request = [&lcf, chunked]() {
             auto cln = http::experimental::client(std::make_unique<loopback_http_factory>(lcf), 1, http::experimental::client::retry_requests::no);
             size_t content_length = 0;
             for (auto _ [[maybe_unused]] : {1, 2}) {
                 auto req = http::request::make("GET", "test", "/test");
                 auto make_request = cln.make_request(
                     std::move(req),
-                    [&content_length](const http::reply& resp, input_stream<char>&& in) {
-                        content_length = resp.content_length;
+                    [&content_length, chunked](const http::reply& resp, input_stream<char>&& in) {
+                        content_length = chunked ? 128_KiB : resp.content_length;
```

Comment:
Where does this 128KB come from? Response chunks have a length written by the server - the client shouldn't guess them. I see below that your server indeed returns the entire response as one chunk - and writes the chunk size in the beginning of the chunk.